[POS][Custom payment UI] – Remove "collect payment" button

samiuelson opened the pull request:
#### Remove "collect payment" button and start the payment flow automatically

<!-- Remember about a good descriptive title. -->

Closes: #12828
<!-- Id number of the GitHub issue this PR addresses. -->

### Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->

The goal of this PR is to remove "Collect payment" button in POS Totals screen, and start the payment flow automatically.

This PR covers only a happy path, with the following assumptions:
* Reader is connected
* Order is created successfully

Unhappy paths will be handled in the next PRs.

⚠️ Don't merge until the base branch is not merged into trunk.

### Steps to reproduce
<!-- Step-by-step testing instructions. For new user flows, consider instead stating the goal of the workflow and see if your PR reviewer can accomplish the workflow without specific steps! -->
1. Go to Woo POS (Menu more > Point of Sale)
2. Ensure reader is connected
3. Add items to cart and click "Checkout"
4. After order draft is created, tap the card on the reader to collect payment
5. Verify "Payment success" screen is shown

### Testing information
<!-- This is your opportunity to break out individual scenarios that need testing (when necessary) and/or include a checklist for the reviewer to go through. Consider documenting the following from your own completed testing: devices used, alternate workflows, edge cases, affected areas, critical flows, areas not tested, and any remaining unknowns. Provide feedback on this new section of the PR template through Sept 30, 2024 to Apps Quality; additional context here: p91TBi-b8z-p2#comment-12036 -->
It should be verified that it's possible to collect card payment without the "collect payment" button, assuming happy path conditions are met.

### The tests that have been performed
<!-- To give the reviewer an idea of what could be missed in terms of testing -->
I verified it's possible to collect card payment without the button (see video below).

### Images/gif
<!-- Include before and after images or gifs when appropriate. -->

https://github.com/user-attachments/assets/654048de-117e-4568-8589-625e80b17bbd



- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [ ] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [ ] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [ ] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on big (tablet) and small (phone) in case of UI changes, and no regressions are added.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

samiuelson committed:
Restrict cancelable states by common interface

To be able to filter them using when expression.

samiuelson committed:
Attempt to cancel a payment on back press

samiuelson committed:
Revert "Restrict cancelable states by common interface"

This reverts commit a02047d10024b546a79410a8a2445422514cac5b.

samiuelson committed:
Remove "collect payment" button

Start the flow automatically

samiuelson committed:
Use custom, cancelable payment scope

samiuelson committed:
Update paymentScope to inherit from viewModel scope

samiuelson committed:
Remove unused imports

dangermattic commented:
<!--
  1 Error: PR requires at least one label...
  0 Warnings
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">PR requires at least one label.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

samiuelson committed:
Remove redundant `onClear` implementation

AnirudhBhat reviewed (approved):
LGTM :shipit:

samiuelson committed:
Manage payment coroutine scope internally in controller

samiuelson committed:
Clean up code

samiuelson committed:
Remove unused code

I've checked the history, and `CardReaderPaymentViewModel::reFetchOrder` has never been invoked outside of the class. When extracting payment flow to the controller, I missed the `@VisibleForTesting` annotation and treated it as a public function when it was supposed to be treated as `private`. I'm removing this method from VM – it's supposed to be invoked internally in `CardReaderPaymentController`.

samiuelson committed:
Remove delay(1) from tests

samiuelson committed:
Merge pull request #13037 from woocommerce/custom-payment-ui-7-payment-cancelation

[POS][Custom payment UI] – Payment cancelation
